File: Algoritms/Csharp/ClassTest/AllSortingTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Csharp.Sorting.Quadratic;
using Csharp.Sorting.NLogN;
using Csharp.Sorting.LinearyTime;
using Csharp.Sorting;

namespace ClassTest
{
    [TestClass]
    public class AllSortingTest
    {
        private BubbleSort<int> bbs = new BubbleSort<int>();
        private InsertionSort<int> ins = new InsertionSort<int>();
        private SelectionSort<int> sns = new SelectionSort<int>();
        private QuickSort<int> qks = new QuickSort<int>();
        private MergeSort<int> mgs = new MergeSort<int>();
        private CountingSort cgs = new CountingSort();
        private int[] array = Util.RandomList(10000);


        public bool isOrdened(int[] array)
        {
            bool ans = true;

            for(int i = 1; i< array.Length; i++)
            {
                if(array[i] < array[i - 1])
                {
                    ans = false;
                    break;
                }
            }
            return ans;
        }

        


        [TestMethod]
        public void BubbleSortTest()
        {
            Assert.IsFalse(this.isOrdened(array));
            bbs.Sort(array);
            Assert.IsTrue(this.isOrdened(array));

        }

        [TestMethod]
        public void SelectionSortTest()
        {
            Assert.IsFalse(this.isOrdened(array));
            sns.Sort(array);
            Assert.IsTrue(this.isOrdened(array));

        }

        [TestMethod]
        public void InsertionSortTest()
        {
            Assert.IsFalse(this.isOrdened(array));
            ins.Sort(array);
            Assert.IsTrue(this.isOrdened(array));

        }

        [TestMethod]
        public void MergeSortTest()
        {
            Assert.IsFalse(this.isOrdened(array));
            mgs.Sort(array);
            Assert.IsTrue(this.isOrdened(array));

        }

        [TestMethod]
        public void QuickSortTest()
        {
            Assert.IsFalse(this.isOrdened(array));
            qks.Sort(array);
            Assert.IsTrue(this.isOrdened(array));

        }

        [TestMethod]
        public void CountingSortTest()
        {
            Assert.IsFalse(this.isOrdened(array));
            cgs.Sort(array);
            Assert.IsTrue(this.isOrdened(array));

        }
    }
}

File: Algoritms/Java/Sorting/TestClass/AllSortingTest.java
package TestClass;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import LinearyTime.CountingSort;
import NLogN.MergeSort;
import NLogN.QuickSort;
import Quadratic.BubbleSort;
import Quadratic.InsertionSort;
import Quadratic.SelectionSort;
import Util.Util;

public class AllSortingTest {

	private BubbleSort<Integer> bbs = new BubbleSort<>();
	private InsertionSort<Integer> ins = new InsertionSort<>();
	private SelectionSort<Integer> sns = new SelectionSort<>();
	private QuickSort<Integer> qks = new QuickSort<>();
	private MergeSort<Integer> mgs = new MergeSort<>();
	private CountingSort cgs = new CountingSort();
	private Integer[] array = Util.RandomList(30);

	private boolean isOrdened(Integer[] array) {
		boolean ans = true;

		for (int i = 1; i < array.length; i++) {
			if (array[i] < array[i - 1]) {
				ans = false;
				break;
			}
		}
		return ans;
	}

	@Test
	public void BubbleSortTest() {
		assertFalse(this.isOrdened(array));
		bbs.Sort(array);
		assertTrue(this.isOrdened(array));

	}

	@Test
	public void SelectionSortTest() {
		assertFalse(this.isOrdened(array));
		sns.Sort(array);
		assertTrue(this.isOrdened(array));

	}

	@Test
	public void InsertionSortTest() {
		assertFalse(this.isOrdened(array));
		ins.Sort(array);
		assertTrue(this.isOrdened(array));

	}

	@Test
	public void MergeSortTest() {
		assertFalse(this.isOrdened(array));
		mgs.Sort(array);
		assertTrue(this.isOrdened(array));

	}

	@Test
	public void QuickSortTest() {
		assertFalse(this.isOrdened(array));
		qks.Sort(array);
		assertTrue(this.isOrdened(array));

	}

	@Test
	public void CountingSortTest() {
		assertFalse(this.isOrdened(array));
		cgs.Sort(array);
		assertTrue(this.isOrdened(array));

	}

}
